korman/properties/modifiers/gui.py

```diff
@@ -330,3 +331,249 @@ def _create_moul_nodes(self, clickable_object, nodes, journalnode, age_name):
+                             description="Type of Linking Animation to use",
+                             items=[("LinkOut", "Standing", "The avatar steps up to the book and places their hand on the panel"),
+                                    ("FishBookLinkOut", "Kneeling", "The avatar kneels in front of the book and places their hand on the panel"),],
+                             default="LinkOut",
+                             options=set())
+
+    # Interactables
+    seek_point = PointerProperty(name="Seek Point",
+                                 description="The point the avatar will seek to before opening the Linking Book GUI",
+                                 type=bpy.types.Object,
+                                 poll=idprops.poll_empty_objects)
+    clickable_region = PointerProperty(name="Clickable Region",
+                                 description="The region in which the avatar must be standing before they can click on the Linking Book",
+                                 type=bpy.types.Object,
+                                 poll=idprops.poll_mesh_objects)
+    clickable = PointerProperty(name="Clickable",
+                                 description="The object the avatar will click on to activate the Linking Book GUI",
+                                 type=bpy.types.Object,
+                                 poll=idprops.poll_mesh_objects)
+
+    # -- Path of the Shell options --
+    # Popup Appearance
+    book_scale_w = IntProperty(name="Book Width Scale",
+                               description="Width scale",
+                               default=100, min=0, max=100,
+                               subtype="PERCENTAGE")
+    book_scale_h = IntProperty(name="Book Height Scale",
+                               description="Height scale",
+                               default=100, min=0, max=100,
+                               subtype="PERCENTAGE")
+    # TODO: link image
+    # TODO: stamp texture
+    stamp_x = IntProperty(name="Stamp Position X",
+                          description="Width scale",
+                          default=100, min=0, max=100,
+                          subtype="PERCENTAGE")
+    stamp_y = IntProperty(name="Stamp Position Y",
+                          description="Height scale",
+                          default=100, min=0, max=100,
+                          subtype="PERCENTAGE")
+
+    def export(self, exporter, bo, so):
+        our_versions = (globals()[j] for j in self.versions)
+        version = exporter.mgr.getVer()
+        if version not in our_versions:
+            # We aren't needed here
+            exporter.report.port("Object '{}' has a LinkingBookMod not enabled for export to the selected engine.  Skipping.",
+                                 bo.name, version, indent=2)
+            return
+
+
+        if self.clickable is None:
+            raise ExportError("{}: Linking Book modifier requires a clickable!".format(bo.name))
```

Comment:
No need to format an export error.

Reply:
Except that it's really useful having the error actually tell you which object caused the export failure.

Reply:
Clarification: pass `bo.name` as an argument to `ExportError`--it applies `format()` for you.

Reply:
Ah, I see!